compiler/sigmatch.nim

```diff
@@ -1362,18 +1401,20 @@ proc typeRel(c: var TCandidate, f, aOrig: PType,
             result = isNone
 
   of tyPtr, tyRef:
-    skipOwned(a)
-    if a.kind == f.kind:
+    #skipOwned(a)
+    let effectiveArgType = getObjectTypeOrNil(a)
```

Comment:
Neither the newly added tests nor the specification nor logic suggest that the handling of `ptr` and `ref` need to care about `getObjectTypeOrNil`.

Reply:
As stated in my comment there are "alignment issues" with `ptr` and `ref`, but that only applied to my other PR about generic type variables. It seems evident that this is not the correct way to handle that issue anyway, so I'll just take note of this and make it a point to implement it the correct way in that PR.